### PR TITLE
Update metrics top comments and structure

### LINKED ISSUE #124: Label keys/values in diagram in metrics.proto do not match the proto specification
The [diagram](https://github.com/open-telemetry/opentelemetry-proto/blob/master/opentelemetry/proto/metrics/v1/metrics.proto#L46-L77) in metrics.proto does not match the specification.
Diagram indicates that Descriptor contains only Keys while the proto say it contains [keys and values](https://github.com/open-telemetry/opentelemetry-proto/blob/d496c80b353bc4a4f754ae686b59ca3c41de0946/opentelemetry/proto/metrics/v1/metrics.proto#L165-L167) that are common to all data points.

### LINKED ISSUE #177: Update MetricDescriptor comment to reflect the label key-value structure properly
The comment on MetricDescriptor currently states that label keys are part of the MetricDescripror, while label values are defined in each data point: https://github.com/open-telemetry/opentelemetry-proto/blob/c44400dc2b039380383b4fa4f243c49d7baf83c1/opentelemetry/proto/metrics/v1/metrics.proto#L60-L61

This has been changed in #102 to each data point having key-value pairs for labels: https://github.com/open-telemetry/opentelemetry-proto/blob/c44400dc2b039380383b4fa4f243c49d7baf83c1/opentelemetry/proto/metrics/v1/metrics.proto#L245-L246, so this comment needs to be updated as well.

/cc @bogdandrutu

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes #177
Fixes #124

This PR is extracted from #168 and a huge thanks goes to @jmacd. It is updated for the current state, but I heard more people complaining about this and thought it is good to fix now.

### CODE REVIEW

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 1]
```diff
@@ -46,20 +46,24 @@ message InstrumentationLibraryMetrics {
 
 // Defines a Metric which has one or more timeseries.
 //
-// The data model and relation between entities is shown in the diagram below.
+// The data model and relation between entities is shown in the
+// diagram below. Here, "DataPoint" is the term used to refer to any
+// one of the specific data point kinds, and "points" is the term use
```
Comment (cijothomas) [inline opentelemetry/proto/metrics/v1/metrics.proto:51]: "kind" is not defined anywhere. Did you mean Temporality?

Comment (cijothomas) [inline opentelemetry/proto/metrics/v1/metrics.proto:51]: use -> used

Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto:51]: yeah, wrong copy paste from the other PR. Updated.

Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto:51]: Good call, thanks.

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 2]
```diff
@@ -86,13 +90,15 @@ message InstrumentationLibraryMetrics {
...
-// DataPoint is a value of specific type corresponding to a given moment in
-// time. Each DataPoint is timestamped.
-//
-// DataPoint is strongly typed: each DataPoint type has a specific Protobuf message
-// depending on the value type of the metric and thus there are currently 4 DataPoint
-// messages, which correspond to the types of metric values.
+// All DataPoint types have includes three common fields:
+// - Labels are the optional key-value pairs associated with the data point.
+// - StartTimeUnixNano MUST be set to the start of the interval when the
+//   descriptor kind includes CUMULATIVE or DELTA.  This field is not set
```
Comment (cijothomas) [inline opentelemetry/proto/metrics/v1/metrics.proto:96]: kind -> Temporality

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 3]
```diff
@@ -86,13 +90,15 @@ message InstrumentationLibraryMetrics {
...
 //                       +---------------------------+
 //
-//-----------------------------------------------------------------------
-// DataPoint is a value of specific type corresponding to a given moment in
-// time. Each DataPoint is timestamped.
-//
-// DataPoint is strongly typed: each DataPoint type has a specific Protobuf message
-// depending on the value type of the metric and thus there are currently 4 DataPoint
-// messages, which correspond to the types of metric values.
+// All DataPoint types have includes three common fields:
```
Comment (cijothomas) [inline opentelemetry/proto/metrics/v1/metrics.proto:93]: This line doesn't sound correct. 
Is that saying "All DataPoint types must include the following three common fields:"

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 4]
```diff
@@ -86,13 +90,15 @@ message InstrumentationLibraryMetrics {
...
+// - Labels are the optional key-value pairs associated with the data point.
```
Comment (cijothomas) [inline opentelemetry/proto/metrics/v1/metrics.proto:94]: If labels are optional, then its incorrect to say DataPoint must include "three" common field.

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 5]
```diff
@@ -86,13 +90,15 @@ message InstrumentationLibraryMetrics {
...
+// - StartTimeUnixNano MUST be set to the start of the interval when the
+//   descriptor kind includes CUMULATIVE or DELTA.  This field is not set
+//   for INSTANTANEOUS timeseries, where instead the TimeUnixNano field is
+//   set for individual points.
+// - TimeUnixNano MUST be set to:
+//   - the end of the interval (CUMULATIVE or DELTA)
+//   - the instantaneous time of the event (INSTANTANEOUS).
```
Comment (cijothomas) [inline opentelemetry/proto/metrics/v1/metrics.proto:101]: what about the actual metric value itself?

File: opentelemetry/proto/metrics/v1/metrics.proto (hunk 2 above)
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto]: Thanks.

File: opentelemetry/proto/metrics/v1/metrics.proto (hunk 4 above)
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto]: Corrected "zero or more" is not optional. Zero is accepted.

File: opentelemetry/proto/metrics/v1/metrics.proto (hunk 3 above)
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto]: just removed "includes".

**Comment by bogdandrutu** [comment]
Thanks for the review @cijothomas

**Pull request merged**